ractor/src/actor/supervision.rs

```diff
@@ -134,15 +134,21 @@ impl SupervisionTree {
         }
     }
 
-    pub(crate) fn get_children(&self) -> Vec<ActorCell> {
+    pub(crate) fn take_children(&self) -> Vec<ActorCell> {
```

Comment:
we shouldn't need a take method anywhere, the supervision tree should only be modified upon link/unlink. This should be as easy as just replacing the functionality inside with the non-clearing version, and updating a test case to make sure that the supervision tree isn't modified.

#310 is indeed a bug, and fixing it vs adding a new API like this is better for footgun safety.

Reply:
I think some kind of clearing is still required. Initially I just removed the `clear()` but then the double-link test run out of stack. I realized if we don't clear the children list after unlink, we can run into mutual recursion in the double-link scenario. So instead of calling `clear()` everywhere I created the take method.